### ISSUE TITLE
Uncaught Callback must be a function

### ISSUE THREAD

**Issue by gurius** [jakesankey/git-history#52]
[Enter steps to reproduce:]

1. Call "Show File History" on typescript file
2. Choose one commit from list

**Atom**: 1.34.0 x64
**Electron**: 3.1.4
**OS**: "Arch Linux"
**Thrown From**: [git-history](https://github.com/jakesankey/git-history) package 3.3.0

### Stack Trace

Uncaught Callback must be a function

```
At fs.js:165

TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
    at makeCallback (fs.js:165:11)
    at Object.fs.mkdir (fs.js:854:14)
    at /packages/git-history/lib/git-history-view.coffee:104:20
    at triggerExitCallback (/usr/lib/atom/src/buffered-process.js:258:9)
    at bufferStream (/usr/lib/atom/src/buffered-process.js:271:9)
    at Socket.stream.on (/usr/lib/atom/src/buffered-process.js:155:7)
    at Socket.emit (events.js:187:15)
    at Pipe._handle.close [as _onclose] (net.js:596:12)
```

### Commands

```
     -6:02.6.0 core:save (input.hidden-input)
     -6:01.7.0 core:backspace (input.hidden-input)
     -6:01.5.0 core:save (input.hidden-input)
     -5:54.3.0 editor:join-lines (input.hidden-input)
     -5:53.3.0 core:undo (input.hidden-input)
     -5:51.4.0 git-history:show-file-history (input.hidden-input)
     -5:48.1.0 core:move-down (input.hidden-input)
     -5:47.5.0 core:move-up (input.hidden-input)
     -5:47.2.0 core:confirm (input.hidden-input)
     -5:45.3.0 editor:consolidate-selections (input.hidden-input)
     -5:45.3.0 core:cancel (input.hidden-input)
     -5:29 git-history:show-file-history (input.hidden-input)
     -5:28.2.0 core:move-down (input.hidden-input)
     -5:27.9.0 core:confirm (input.hidden-input)
     -1:04.1.0 git-history:show-file-history (input.hidden-input)
     -1:02.6.0 core:confirm (input.hidden-input)
```

### Non-Core Packages

```
angular-2-typeScript-snippets 0.7.0 
angular-snippets 7.0.5 
ariake-dark-syntax 0.1.3 
ask-stack 2.2.0 
atom-bracket-highlight 0.3.2 
atom-dark-fusion-syntax 2.2.1 
atom-hipster-ui 0.3.0 
atom-ide-debugger-node 0.7.3 
atom-ide-ui 0.13.0 
atom-material-syntax 1.0.8 
atom-material-syntax-dark 1.0.0 
atom-material-syntax-light 0.4.6 
atom-material-ui 2.1.3 
atom-meld 0.3.5 
atom-monokai 0.10.9 
atom-monokai-flat 3.0.0 
atom-slack-snippets 0.2.0 
atom-typescript 13.1.0 
base16-atelierdune-dark-syntax 2.0.3 
batman-syntax 1.1.0 
brogrammer-syntax 1.0.0 
brogrammer-ui 1.0.0 
bumblebee-syntax 1.0.0 
charcode-display 0.2.1 
chester-atom-2-syntax 0.4.0 
chester-atom-syntax 0.3.2 
code-peek 1.4.21 
color-picker 2.3.0 
dash-ui 0.4.4 
dimmed-monokai 0.1.0 
docblockr 0.13.7 
duotone-dark-forest-syntax 2.1.0 
easy-monokai-deuce-syntax 0.3.0 
editorconfig 2.3.0 
emmet 2.4.3 
es6-javascript 1.0.0 
fancy-bracket-matcher 2.1.0 
file-icons 2.1.30 
ganymede-atom-syntax 1.1.1 
git-blame 1.8.0 
git-history 3.3.0 
git-time-machine 2.1.0 
glowing-cursor 0.1.2 
gruvbox-plus-syntax 1.3.1 
highlight-selected 0.14.0 
hipster-syntax 0.0.3 
hydra-syntax-theme 4.0.1 
ide-typescript 0.8.0 
improved-chester-atom-syntax 0.3.0 
intellij-darcula-syntax 1.0.11 
intentions 1.1.5 
js-refactor 0.8.1 
kimbie-dark 2.0.2 
language-diff 0.7.0 
language-mongodb 0.3.1 
language-patch 1.2.0 
language-regexp 1.0.1 
material-ui 1.0.10 
minimap 4.29.9 
minimap-git-diff 4.3.1 
minimap-selection 4.5.0 
minimap-split-diff 0.3.7 
monokai-flat 1.0.1 
monokai-flat-ui 4.2.0 
monokai-neo-syntax 1.3.2 
open-in-browser 0.5.2 
pastel-jelly-ui 0.3.0 
pen-paper-coffee-syntax 0.17.0 
pigments 0.40.2 
pinned-tabs 2.1.1 
pristine-ui 1.1.11 
redeye-ui 0.5.0 
refactor 0.11.5 
regex-railroad-diagram 0.19.4 
simplyy-snippets 0.4.16 
split-diff 1.6.0 
steam-pirate-ui 0.6.1 
svg-preview 0.14.0 
sync-settings 0.8.6 
twilight-syntax-theme 0.1.1 
Warm-Dark-syntax 0.1.0 
```

**Comment by racosa** [comment]
same problem here... does someone knows how to fix it?

**Comment by aastrong** [comment]
Same issue here.

**Comment by kielnino** [comment]
I also have the exact same issue

**Comment by aastrong** [comment]
> Same issue here.

To follow up on my own comment - the commit wit the fix from @racosa doesnt seem to fix anything for myself.

**Comment by ybenitezf** [comment]
`+ 1`

**Comment by oO0oO0oO0o0o00** [comment]
This happens for every file, not only `typescript`s, which made the package completely useless at the time.

**Comment by alejoberman-accor** [comment]
Same problem

**Comment by alejoberman** [comment]
Same problem, every file, every commit

**Comment by ovanderzee** [comment]
End of service-life. Sadly most history trackers for Atom are two years old or older. I found git-time-machine. It is not flawless, but it works. Good suggestions, anybody?

**Comment by endro** [comment]
Creating an empty .git-history directory in Atom config dir (for me ~/.atom/.git-history) solves the problem.

**Comment by oO0oO0oO0o0o00** [comment]
> Creating an empty .git-history directory in Atom config dir (for me ~/.atom/.git-history) solves the problem.

... How did you find this out, meow~

**Comment by rgiannico** [comment]
> Creating an empty .git-history directory in Atom config dir (for me ~/.atom/.git-history) solves the problem.

great hack @endro !!! It works perfectly for me! 
In windows I just created this empty folder   `C:\Users\myusername\.atom\.git-history`

**Comment by bob-steffen** [comment]
@endro excellent fix! works for me.

**Comment by pheki** [comment]
@gurius can you reopen this issue?

This hasn't fixed, it still happens on new installations (which must manually create the file) and will only be fixed for everyone if #55 ever lands
